### PR TITLE
[Security] Build and run with Node 16

### LINKED ISSUE #1: Upgrade to supported Node version
Switch to node:4, node:6, or node:8 in [Dockerfile](https://github.com/zooniverse/markdown-api/blob/master/Dockerfile#L1).

### PR DESCRIPTION
**Pull request by eatyourgreens**
- Convert the Dockerfile to `node16-alpine`.
- Convert `package-lock.json` to version 2.
- Install with `npm ci`.

- Closes #1.

### CODE REVIEW

**Comment by eatyourgreens** [comment]
I tested this locally with `docker compose build` and `docker compose up` and [the example requests](https://github.com/zooniverse/markdown-api#examples). Everything worked, but I'll wait on a second pair of eyes before merging.

**Review by camallen — APPROVED** [review]
LGTM

**Pull request merged**
